### PR TITLE
Add some pitfalls to the dotnet crypto code

### PR DESCRIPTION
**Pull request by tghosth**

### CODE REVIEW

**Comment by tghosth** [comment]
Waiting for #1158 to be merged

File: cheatsheets/DotNet_Security_Cheat_Sheet.md [hunk 1]
```diff
@@ -51,10 +53,21 @@ The .NET Framework is the set of APIs that support an advanced type system, data
 - Use the [Windows Data Protection API (DPAPI)](https://docs.microsoft.com/en-us/dotnet/standard/security/how-to-use-data-protection) for secure local storage of sensitive data.
 - Where DPAPI cannot be used, follow the algorithm guidance in the [OWASP Cryptographic Storage Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html#algorithms).
 
-The following code snippet shows an example of using AES-GCM to perform encryption/decryption of data.
+The following code snippet shows an example of using AES-GCM to perform encryption/decryption of data. It is strongly recommended to have a cryptography expert review your final design and code, as even the most trivial error can severely weaken your encryption.
 
 The code is based on example from here: [https://www.scottbrady91.com/c-sharp/aes-gcm-dotnet](https://www.scottbrady91.com/c-sharp/aes-gcm-dotnet)
 
+A few contraints/pitfalls with this code:
```
Comment (szh) [inline cheatsheets/DotNet_Security_Cheat_Sheet.md:60]: *constraints

**Review by szh — COMMENTED** [review]
I like how this is looking. Minor typo noted

File: cheatsheets/DotNet_Security_Cheat_Sheet.md (hunk 1 above)
Comment (tghosth) [reply cheatsheets/DotNet_Security_Cheat_Sheet.md]: Whoops, thanks for spotting that!

**Comment by tghosth** [comment]
#1158 got merged so this can now be merged @szh @jmanico

**Comment by tghosth** [comment]
Any chance this can now be merged @szh @jmanico ?

File: cheatsheets/DotNet_Security_Cheat_Sheet.md [hunk 2]
```diff
@@ -130,14 +143,33 @@ public static class AesGcmSimple
 
 ```
 
+</details>
+
+_**End of code snippet.**_
```
Comment (szh) [inline cheatsheets/DotNet_Security_Cheat_Sheet.md:148]: Do we need this? It seems extraneous.

Comment (tghosth) [reply cheatsheets/DotNet_Security_Cheat_Sheet.md:148]: So the problem is that I am not sure how clearly the collapsible sections will render so I want to make sure it is clear where the code starts and ends @szh

Comment (szh) [reply cheatsheets/DotNet_Security_Cheat_Sheet.md:148]: @tghosth Is it not going to look like it does in the Github preview?

Comment (tghosth) [reply cheatsheets/DotNet_Security_Cheat_Sheet.md:148]: So it looks different in the real cheatsheet site to how it looks in Github preview and [actually looks a lot better](https://cheatsheetseries.owasp.org/cheatsheets/Java_Security_Cheat_Sheet.html#symmetric-example-using-google-tink) so I am going to remove the extra comment in a new PR

Comment (tghosth) [reply cheatsheets/DotNet_Security_Cheat_Sheet.md:148]: See #1162 @szh

**Pull request merged**